### PR TITLE
Document Target and TargetData

### PR DESCRIPTION
**Pull request by CodaFi**
There is an inconsistency here between the name of `LLVMSizeOfTypeInBits` and the LLVM documentation.

### CODE REVIEW

File: Sources/LLVMSwift/TargetData.swift [hunk 1]
```diff
@@ -1,139 +1,344 @@
...
+    return Int(LLVMOffsetOfElement(llvm, type.asLLVM(), UInt32(element)))
+  }
+
+  /// Computes the size of a type in bytes for a target.
+  ///
+  /// - parameter type: The type to compute the size of.
+  ///
+  /// - returns: The size of the type in bytes.
+  public func sizeOfTypeInBits(_ type: IRType) -> Int {
+    return Int(LLVMSizeOfTypeInBits(llvm, type.asLLVM()))
```
Comment (CodaFi) [inline Sources/LLVMSwift/TargetData.swift:169]: "Bits, bytes, what's the difference"

- LLVM

Comment (harlanhaskins) [reply Sources/LLVMSwift/TargetData.swift:169]: If LLVM's is named wrong, then maybe we should rename this method...

File: Sources/LLVMSwift/TargetData.swift [hunk 2]
```diff
@@ -1,139 +1,344 @@
...
+    switch self {
+    case .default: return LLVMRelocDefault
+    case .static: return LLVMRelocStatic
+    case .pic: return LLVMRelocPIC
+    case .dynamicNoPIC: return LLVMRelocDynamicNoPic
     }
+  }
 }
 
+/// The model generated code should follow.  This enables particular styles of
```
Comment (harlanhaskins) [inline Sources/LLVMSwift/TargetData.swift]: I think this could benefit from 'that' or 'which'

**Review by harlanhaskins — CHANGES_REQUESTED** [review]

File: Sources/LLVMSwift/TargetData.swift (hunk 1 above)
Comment (CodaFi) [reply Sources/LLVMSwift/TargetData.swift:169]: Trouble is, I'm not entirely sure which is actually correct.

Comment (CodaFi) [reply Sources/LLVMSwift/TargetData.swift:169]: Ah, it's bits.

Comment (harlanhaskins) [reply Sources/LLVMSwift/TargetData.swift:169]: print(module.dataLayout.sizeOfTypeInBits(IntType.int64))
// 64

Comment (CodaFi) [reply Sources/LLVMSwift/TargetData.swift:169]: Sorry, forgot to update the footer text.

**Review by harlanhaskins — APPROVED** [review]

**Comment by harlanhaskins** [comment]
I'll merge it once you run jazzy again

**Pull request merged**